From: Simon McVittie <smcv@debian.org>
Date: Wed, 17 Jul 2019 13:54:41 +0100
Subject: gdk: Don't distribute generated files in tarballs

gdkversionmacros.h is generated by configure, so it should be in
DISTCLEANFILES. The rest of $(gdk_built_sources) are built at compile
time by GLib tools, so in principle they should be in CLEANFILES, but
DISTCLEANFILES is close enough.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 gdk/Makefile.am | 21 +++------------------
 1 file changed, 3 insertions(+), 18 deletions(-)

diff --git a/gdk/Makefile.am b/gdk/Makefile.am
index 525c5a0..64e5b17 100644
--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -203,16 +203,8 @@ nodist_gdkinclude_HEADERS = gdkconfig.h gdkenumtypes.h gdkversionmacros.h
 deprecatedincludedir = $(includedir)/gtk-3.0/gdk/deprecated
 deprecatedinclude_HEADERS = $(deprecated_h_sources)
 
-common_sources = 		\
-	$(gdk_private_headers)	\
-	$(gdk_c_sources)	\
-	gdkenumtypes.c		\
-	gdkmarshalers.c		\
-	gdkmarshalers.h		\
-	gdkresources.h		\
-	gdkresources.c
-
-libgdk_3_la_SOURCES = $(common_sources)
+libgdk_3_la_SOURCES = $(gdk_private_headers) $(gdk_c_sources)
+nodist_libgdk_3_la_SOURCES = $(gdk_built_sources)
 libgdk_3_la_CFLAGS = $(AM_CFLAGS) $(GDK_HIDDEN_VISIBILITY_CFLAGS)
 libgdk_3_la_LIBADD = $(GDK_DEP_LIBS) $(SHM_LIBS)
 libgdk_3_la_LDFLAGS = $(LDADD)
@@ -417,7 +409,7 @@ endif
 
 lib_LTLIBRARIES = libgdk-3.la
 
-MAINTAINERCLEANFILES = $(gdk_built_sources)
+DISTCLEANFILES = $(gdk_built_sources)
 EXTRA_DIST += \
 	$(gdk_built_sources)	\
 	fallback-c89.c
@@ -562,13 +554,6 @@ install-data-local: install-ms-lib install-def-file
 uninstall-local: uninstall-ms-lib uninstall-def-file
 	rm -f $(DESTDIR)$(configexecincludedir)/gdkconfig.h
 
-# if srcdir!=builddir, clean out maintainer-clean files from builddir
-# this allows dist to pass.
-distclean-local:
-	if test $(srcdir) != .; then \
-	  rm -f $(MAINTAINERCLEANFILES); \
-	fi
-
 .PHONY: files
 
 files:
